### DISCUSSION TITLE
httpOnly cookie authentication

### DISCUSSION THREAD

**Discussion by 4lp** [enisdenjo/graphql-ws#530] [Q&A] [answered]
Is there any way to use an `httpOnly` cookie for authentication in graphql-ws? The http graphql library I'm using include a `credentials` option to set cookies for those requests but I haven't be able to find anything similar for this library. Right now I'm having to exact a JWT from a cookie and use the value for the request `authorization` header, meaning I can't set it as `httpOnly`.

**Comment by enisdenjo** [comment, marked as answer]
The browser native WebSocket automatically sends cookies alongside the upgrade request. [There are no credentials configuration options when instantiating a WebSocket.](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket/WebSocket) Also see https://github.com/enisdenjo/graphql-ws/discussions/257#discussioncomment-1561449.

Furthermore, there's no way you can set custom headers in the browser's WebSocket.

[Also beware that `httpOnly` cookies cant be accessed from JavaScript.](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#httponly)

**Reply by 4lp** [reply]
thanks for the quick reply! based on the links you provided it doesn't seem like this is possible so I'll just have to use my current strategy
